File: android/src/main/java/smoksmog/logger/LevelBlockingLogger.java
package smoksmog.logger;

import android.util.Log;

public class LevelBlockingLogger implements Logger {

    private final Logger logger;
    private final int lowestLevelToLog;

    /**
     * @param logger           implementation
     * @param lowestLevelToLog use Log.* value / level, eg. Log.ERROR
     */
    public LevelBlockingLogger(Logger logger, int lowestLevelToLog) {
        this.logger = logger;
        this.lowestLevelToLog = lowestLevelToLog;
    }

    @Override
    public void v(String tag, String message) {
        if (lowestLevelToLog >= Log.VERBOSE) {
            logger.v(tag, message);
        }
    }

    @Override
    public void v(String tag, String message, Throwable throwable) {
        if (lowestLevelToLog >= Log.VERBOSE) {
            logger.v(tag, message, throwable);
        }
    }

    @Override
    public void d(String tag, String message) {
        if (lowestLevelToLog >= Log.DEBUG) {
            logger.d(tag, message);
        }
    }

    @Override
    public void d(String tag, String message, Throwable throwable) {
        if (lowestLevelToLog >= Log.DEBUG) {
            logger.d(tag, message, throwable);
        }
    }

    @Override
    public void i(String tag, String message) {
        if (lowestLevelToLog >= Log.INFO) {
            logger.i(tag, message);
        }
    }

    @Override
    public void i(String tag, String message, Throwable throwable) {
        if (lowestLevelToLog >= Log.INFO) {
            logger.i(tag, message, throwable);
        }
    }

    @Override
    public void w(String tag, String message) {
        if (lowestLevelToLog >= Log.WARN) {
            logger.w(tag, message);
        }
    }

    @Override
    public void w(String tag, String message, Throwable throwable) {
        if (lowestLevelToLog >= Log.WARN) {
            logger.w(tag, message, throwable);
        }
    }

    @Override
    public void e(String tag, String message) {
        if (lowestLevelToLog >= Log.ERROR) {
            logger.e(tag, message);
        }
    }

    @Override
    public void e(String tag, String message, Throwable throwable) {
        if (lowestLevelToLog >= Log.ERROR) {
            logger.e(tag, message, throwable);
        }
    }
}

File: android/src/main/kotlin/smoksmog/logger/LevelBlockingLogger.kt
package smoksmog.logger

import android.util.Log

class LevelBlockingLogger(private val logger: Logger, private val lowestLevelToLog: Int) : Logger {

    override fun v(tag: String, message: String) {
        if (lowestLevelToLog >= Log.VERBOSE) {
            logger.v(tag, message)
        }
    }

    override fun v(tag: String, message: String, throwable: Throwable) {
        if (lowestLevelToLog >= Log.VERBOSE) {
            logger.v(tag, message, throwable)
        }
    }

    override fun d(tag: String, message: String) {
        if (lowestLevelToLog >= Log.DEBUG) {
            logger.d(tag, message)
        }
    }

    override fun d(tag: String, message: String, throwable: Throwable) {
        if (lowestLevelToLog >= Log.DEBUG) {
            logger.d(tag, message, throwable)
        }
    }

    override fun i(tag: String, message: String) {
        if (lowestLevelToLog >= Log.INFO) {
            logger.i(tag, message)
        }
    }

    override fun i(tag: String, message: String, throwable: Throwable) {
        if (lowestLevelToLog >= Log.INFO) {
            logger.i(tag, message, throwable)
        }
    }

    override fun w(tag: String, message: String) {
        if (lowestLevelToLog >= Log.WARN) {
            logger.w(tag, message)
        }
    }

    override fun w(tag: String, message: String, throwable: Throwable) {
        if (lowestLevelToLog >= Log.WARN) {
            logger.w(tag, message, throwable)
        }
    }

    override fun e(tag: String, message: String) {
        if (lowestLevelToLog >= Log.ERROR) {
            logger.e(tag, message)
        }
    }

    override fun e(tag: String, message: String, throwable: Throwable) {
        if (lowestLevelToLog >= Log.ERROR) {
            logger.e(tag, message, throwable)
        }
    }
}
